### PR TITLE
Add restarability to the training

### PR DESCRIPTION
**Pull request by perouz**
Adds the functionality of restarting a job where it was interrupted.

Added the command argument `-restartable`, which when set will resume the job where it was left off. To use this option, you must specify an output directory with `-output_dir`.

### CODE REVIEW

**Comment by perouz** [comment]
1. If you are running a model without specifying an `-output_dir`, it will create an output directory with a random name and save everything there: drawback is that your job cannot be restartable in this case. 
If you want your job to be restartable, you *need to specify an `output_dir`*. If you specify the same output directory for multiple runs, then things *will get overwritten*. 
So to run the same job with multiple parameters, just specify a different output_dir for each and you should be fine.

**Comment by baharefatemi** [comment]
1. This would make running multiple cases hard. I recommend having an output dir in args as the general dir to save and load and then use some naming conventions e.g., lr_filter_w, ... to show what parameters we are looking for. Or something automatic like this without specifying in the args.

**Comment by perouz** [comment]
2. The `test_or_valid` parameter is only used to format the name of the file: If the results are on the test set, then the results (MRR, Hit@10, ....etc) will be saved as a json file named `test_measure_1000itr.json`, otherwise it will be saved with the `valid` prefix. Right now, we are only saving the test results, but we could change this later if needed.

**Comment by perouz** [comment]
1. If we have one general output directory, how would you find out which pretrained model to load? Are we supposed to name the directory with *all* the possible parameter values in the dirname?
On the other hand, it is easy to write code to generate a list of borgy jobs with the right output directory name for each parameter list. 
We need to discuss this further.
